test/settings_test.dart

```diff
@@ -135,5 +149,170 @@ void main() {
+              context: anyNamed('context')))
+          .thenAnswer((_) => Future.value(true));
+
+      await tester.pumpWidget(MultiProvider(providers: [
+        ChangeNotifierProvider<AuthProvider>(create: (_) => mockAuthProvider),
+        ChangeNotifierProvider<WebsiteProvider>(
+            create: (_) => mockWebsiteProvider),
+        ChangeNotifierProvider<QuestionProvider>(
+            create: (_) => mockQuestionProvider),
+        Provider<RequestProvider>(create: (_) => mockRequestProvider),
+      ], child: const MyApp()));
+      await tester.pumpAndSettle();
+
+      // Open settings
+      await tester.tap(find.byIcon(Icons.settings));
+      await tester.pumpAndSettle();
+
+      // Press Ask Permissions page
+      expect(find.text('Request editing permissions'), findsOneWidget);
+      await tester.tap(find.byKey(const ValueKey('ask_permissions')));
+      await tester.pumpAndSettle();
+
+      // Verify nothing happens
+      expect(find.byType(SettingsPage), findsOneWidget);
+    });
+
+    testWidgets('Ask Permissions - user is not verified scenario',
+        (WidgetTester tester) async {
+      when(mockAuthProvider.currentUser).thenAnswer((_) =>
+          Future.value(User(uid: '0', firstName: 'John', lastName: 'Doe')));
+      when(mockAuthProvider.isVerifiedFromCache).thenAnswer((_) => false);
+      when(mockAuthProvider.isAnonymous).thenReturn(false);
+      when(mockRequestProvider.userAlreadyRequested(any,
+              context: anyNamed('context')))
+          .thenAnswer((_) => Future.value(true));
+
+      await tester.pumpWidget(MultiProvider(providers: [
+        ChangeNotifierProvider<AuthProvider>(create: (_) => mockAuthProvider),
+        ChangeNotifierProvider<WebsiteProvider>(
+            create: (_) => mockWebsiteProvider),
+        ChangeNotifierProvider<QuestionProvider>(
+            create: (_) => mockQuestionProvider),
+        Provider<RequestProvider>(create: (_) => mockRequestProvider),
+      ], child: const MyApp()));
+      await tester.pumpAndSettle();
+
+      // Open settings
+      await tester.tap(find.byIcon(Icons.settings));
+      await tester.pumpAndSettle();
+
+      // Press Ask Permissions page
+      expect(find.text('Request editing permissions'), findsOneWidget);
+      await tester.tap(find.byKey(const ValueKey('ask_permissions')));
+
+      // Verify Ask Permissions page is not opened
+      await tester.pumpAndSettle(const Duration(seconds: 4));
+      expect(find.byType(SettingsPage), findsOneWidget);
+
+      // Verify account
+      when(mockAuthProvider.isVerifiedFromCache).thenAnswer((_) => true);
+
+      // Press Ask Permissions page
+      expect(find.text('Request editing permissions'), findsOneWidget);
+      await tester.tap(find.byKey(const ValueKey('ask_permissions')));
+
+      // Verify Ask Permissions page is opened
+      await tester.pumpAndSettle();
+      expect(find.byType(AskPermissions), findsOneWidget);
+    });
```

Comment:
Wrap these in a group called 'Request permissions', and you can remove 'Ask Permissions - ' and the word 'scenario' from the test names (filler word).